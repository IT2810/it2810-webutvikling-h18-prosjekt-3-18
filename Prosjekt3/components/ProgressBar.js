import React, {Component} from 'react';
import {View} from 'react-native';
import {Bar} from 'react-native-progress';

/**
 * Simple Progress bar to be used in Todolists.
 * Height and width should be decided from parent.
 * Library used:
 * https://github.com/oblador/react-native-progress
 */
class ProgressBar extends Component {
    render() {
        let calc_val =
            parseInt(this.props.current)
            / parseInt(this.props.max);
        return (
            <View>
                <Bar
                    progress={calc_val}
                    color={'#0F0'}
                    width={this.props.width}
                    height={this.props.height}
                    unfilledColor={'#F22'}
                    borderColor={"#000"}
                    />
            </View>
        )
    }
}

export default ProgressBar;