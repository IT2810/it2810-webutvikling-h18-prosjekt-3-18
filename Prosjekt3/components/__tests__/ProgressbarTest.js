import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import ProgressBar from '../../components/ProgressBar';

it('renders correctly', () => {
    const tree = renderer.create(<ProgressBar>Snapshot test!</ProgressBar>).toJSON();
    expect(tree).toMatchSnapshot();
});
