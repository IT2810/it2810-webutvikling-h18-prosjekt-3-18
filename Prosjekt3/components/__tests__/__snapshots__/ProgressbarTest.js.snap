// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <View
    animated={true}
    animationConfig={
      Object {
        "bounciness": 0,
      }
    }
    animationType="spring"
    border-radius="12px"
    indeterminate={false}
    onLayout={[Function]}
    progress={NaN}
    style={
      Array [
        Object {
          "backgroundColor": "#F22",
          "borderColor": "#000",
          "borderRadius": 4,
          "borderWidth": 1,
          "overflow": "hidden",
          "width": 150,
        },
        undefined,
      ]
    }
    useNativeDriver={false}
  >
    <View
      style={
        Object {
          "backgroundColor": "#0F0",
          "height": 6,
          "transform": Array [
            Object {
              "translateX": 0,
            },
            Object {
              "translateX": NaN,
            },
            Object {
              "scaleX": NaN,
            },
          ],
        }
      }
    />
  </View>
</View>
`;
